### PR TITLE
[WIP] Rename all files from redis* to valkey*

### PR DESCRIPTION
**Pull request by gaod**
Rename all files from `redis*` to `valkey*` and fixes related code changes causing compilation issues.

### CODE REVIEW

**Comment by mattsta** [comment]
Looks like a clean replace.

There are still references to product names, but not sure if that is getting cleaned up elsewhere, example:

> is the official C client library for Redis. It is used by valkey-cli, valkey-benchmark and Redis Sentinel. It is part of the Redis official ecosystem but is developed externally from the Redis repository, so we just upgrade it as needed.

File: deps/hiredis/adapters/redismoduleapi.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef __HIREDIS_REDISMODULEAPI_H__
 #define __HIREDIS_REDISMODULEAPI_H__
 
-#include "redismodule.h"
+#include "valkeymodule.h"
```
Comment (zuiderkwast) [inline deps/hiredis/adapters/redismoduleapi.h]: Please don't change this. Hiredis is a dependency copied from another repository, so we want to fix it upstream, maybe fork hiredis. This file isn't used by valkey btw.

```suggestion
#include "redismodule.h"
```

File: deps/hiredis/examples/example-redismoduleapi.c [hunk 2]
```diff
@@ -52,7 +52,7 @@ void disconnectCallback(const redisAsyncContext *c, int status) {
 /*
  * This example requires Redis 7.0 or above.
  *
- * 1- Compile this file as a shared library. Directory of "redismodule.h" must
+ * 1- Compile this file as a shared library. Directory of "valkeymodule.h" must
```
Comment (zuiderkwast) [inline deps/hiredis/examples/example-redismoduleapi.c]: Please don't change this. Hiredis is a vendored dependency.

```suggestion
 * 1- Compile this file as a shared library. Directory of "redismodule.h" must
```

File: src/module.c [hunk 3]
```diff
@@ -439,17 +439,17 @@ typedef struct RedisModuleKeyOptCtx {
 } RedisModuleKeyOptCtx;
 
 /* Data structures related to redis module configurations */
-/* The function signatures for module config get callbacks. These are identical to the ones exposed in redismodule.h. */
+/* The function signatures for module config get callbacks. These are identical to the ones exposed in valkeymodule.h. */
```
Comment (zuiderkwast) [inline src/module.c]: Let's not change this filename just now. redismodule.h defines all the structs and functions with names like RedisModuleXXXX.

File: src/module.c [hunk 4]
```diff
@@ -439,17 +439,17 @@ typedef struct RedisModuleKeyOptCtx {
...
 typedef RedisModuleString * (*RedisModuleConfigGetStringFunc)(const char *name, void *privdata);
 typedef long long (*RedisModuleConfigGetNumericFunc)(const char *name, void *privdata);
 typedef int (*RedisModuleConfigGetBoolFunc)(const char *name, void *privdata);
 typedef int (*RedisModuleConfigGetEnumFunc)(const char *name, void *privdata);
-/* The function signatures for module config set callbacks. These are identical to the ones exposed in redismodule.h. */
+/* The function signatures for module config set callbacks. These are identical to the ones exposed in valkeymodule.h. */
```
Comment (zuiderkwast) [inline src/module.c]: Please don't change this just yet.

```suggestion
/* The function signatures for module config set callbacks. These are identical to the ones exposed in redismodule.h. */
```

File: src/module.c [hunk 5]
```diff
@@ -439,17 +439,17 @@ typedef struct RedisModuleKeyOptCtx {
...
 typedef int (*RedisModuleConfigSetStringFunc)(const char *name, RedisModuleString *val, void *privdata, RedisModuleString **err);
 typedef int (*RedisModuleConfigSetNumericFunc)(const char *name, long long val, void *privdata, RedisModuleString **err);
 typedef int (*RedisModuleConfigSetBoolFunc)(const char *name, int val, void *privdata, RedisModuleString **err);
 typedef int (*RedisModuleConfigSetEnumFunc)(const char *name, int val, void *privdata, RedisModuleString **err);
-/* Apply signature, identical to redismodule.h */
+/* Apply signature, identical to valkeymodule.h */
```
Comment (zuiderkwast) [inline src/module.c]:
```suggestion
/* Apply signature, identical to redismodule.h */
```

File: src/module.c [hunk 6]
```diff
@@ -786,7 +786,7 @@ int moduleDelKeyIfEmpty(RedisModuleKey *key) {
  *
  * Note that all the exported APIs are called RM_<funcname> in the core
  * and RedisModule_<funcname> in the module side (defined as function
- * pointers in redismodule.h). In this way the dynamic linker does not
+ * pointers in valkeymodule.h). In this way the dynamic linker does not
```
Comment (zuiderkwast) [inline src/module.c]:
```suggestion
 * pointers in redismodule.h). In this way the dynamic linker does not
```

File: src/module.c [hunk 7]
```diff
@@ -797,7 +797,7 @@ int RM_GetApi(const char *funcname, void **targetPtrPtr) {
      * named API, otherwise REDISMODULE_OK.
      *
      * This function is not meant to be used by modules developer, it is only
-     * used implicitly by including redismodule.h. */
+     * used implicitly by including valkeymodule.h. */
```
Comment (zuiderkwast) [inline src/module.c]:
```suggestion
     * used implicitly by including redismodule.h. */
```

File: src/module.c [hunk 8]
```diff
@@ -11650,7 +11650,7 @@ int RM_SubscribeToServerEvent(RedisModuleCtx *ctx, RedisModuleEvent event, Redis
     /* Protect in case of calls from contexts without a module reference. */
     if (ctx->module == NULL) return REDISMODULE_ERR;
     if (event.id >= _REDISMODULE_EVENT_NEXT) return REDISMODULE_ERR;
-    if (event.dataver > moduleEventVersions[event.id]) return REDISMODULE_ERR; /* Module compiled with a newer redismodule.h than we support */
+    if (event.dataver > moduleEventVersions[event.id]) return REDISMODULE_ERR; /* Module compiled with a newer valkeymodule.h than we support */
```
Comment (zuiderkwast) [inline src/module.c]:
```suggestion
    if (event.dataver > moduleEventVersions[event.id]) return REDISMODULE_ERR; /* Module compiled with a newer redismodule.h than we support */
```

File: src/valkeyassert.h [hunk 9]
```diff
@@ -1,4 +1,4 @@
-/* redisassert.h -- Drop in replacements assert.h that prints the stack trace
+/* valkeyassert.h -- Drop in replacements assert.h that prints the stack trace
```
Comment (zuiderkwast) [inline src/valkeyassert.h]: Let's call this file `serverassert.h` instead. It defines the functions called `_serverAssert` and `_serverPanic`.

**Review by zuiderkwast — CHANGES_REQUESTED** [review]
Hi! Thanks for the PR.

I see you have done these changes in separate commits. That's good because I have to ask you to skip some of them. :)

Everything else LGTM.

**Comment by zuiderkwast** [comment]
Regarding redis-trip, it's deprecated since long ago. It's better that we don't touch it, so please skip this commit too. We plan to keep compatibility with the redis binaries with symlinks, so the script can still work. In the next major release we can delete redis-trib.

**Comment by zuiderkwast** [comment]
CI failure

```
!!! WARNING The following tests failed:

*** [err]: FUNCTION - redis version api in tests/unit/functions.tcl
Expected '255.255.255' to be equal to '' (context: type eval line 15 cmd {assert_equal  [r fcall get_version_v1 0] [r fcall get_version_v2 0]} proc ::test)
```

**Comment by gaod** [comment]
Thanks, I'll take a look about CI failed & the rest of suggestions.

**Comment by zuiderkwast** [comment]
@gaod OK, good.

It's probably easier to revert the commits "Move redismodule.h to valkeymodule.h" and "Move redis-trib to valkey-trib" instead of using my suggestions.

**Comment by hwware** [comment]
Becuase there are thousand of lines in this pr, could you please run a dialy CI and post the result here.  Thanks

File: utils/req-res-log-validator.py [hunk 10]
```diff
@@ -26,7 +26,7 @@
 1. Goes over req-res files, generated by redis-servers, spawned by the testsuite (see logreqres.c)
 2. For each request-response pair, it validates the response against the request's reply_schema (obtained from COMMAND DOCS)
 
-This script spins up a redis-server and a redis-cli in order to obtain COMMAND DOCS.
```
Comment (hwware) [inline utils/req-res-log-validator.py]: redis-server need to be changed

File: utils/req-res-log-validator.py [hunk 11]
```diff
@@ -26,7 +26,7 @@
 1. Goes over req-res files, generated by redis-servers, spawned by the testsuite (see logreqres.c)
```
Comment (hwware) [inline utils/req-res-log-validator.py]: redis-server need to be changed

File: utils/req-res-log-validator.py [hunk 12]
```diff
@@ -273,7 +273,7 @@ def fetch_schemas(cli, port, args, docs):
     parser = argparse.ArgumentParser()
     parser.add_argument('--server', type=str, default='%s/redis-server' % srcdir)
```
Comment (hwware) [inline utils/req-res-log-validator.py]: redis-server need to be changed

File: utils/redis_init_script [hunk 13]
```diff
@@ -13,7 +13,7 @@
 
 REDISPORT=6379
 EXEC=/usr/local/bin/redis-server
```
Comment (hwware) [inline utils/redis_init_script]: redis-server need to be changed

File: utils/create-cluster/create-cluster [hunk 14]
```diff
@@ -132,7 +132,7 @@ fi
 
 echo "Usage: $0 [start|create|stop|restart|watch|tail|tailall|clean|clean-logs|call]"
 echo "start       -- Launch Redis Cluster instances."
-echo "create [-f] -- Create a cluster using redis-cli --cluster create."
+echo "create [-f] -- Create a cluster using valkey-cli --cluster create."
 echo "stop        -- Stop Redis Cluster instances."
```
Comment (hwware) [inline utils/create-cluster/create-cluster:136]: Redis Cluster need to be changed

File: utils/create-cluster/create-cluster [hunk 15]
```diff
@@ -132,7 +132,7 @@ fi
...
 echo "restart     -- Restart Redis Cluster instances."
```
Comment (hwware) [inline utils/create-cluster/create-cluster:137]: Same as above

**Comment by hwware** [comment]
@gaod I found many redis workds do not change to redis, include redis-server, redis cluster etc, could you please check all your changed files again and update them? Thanks

File: src/valkeyassert.h (hunk 9 above)
Comment (daniel-house) [reply src/valkeyassert.h]: +1

**Comment by gaod** [comment]
> @gaod I found many redis workds do not change to redis, include redis-server, redis cluster etc, could you please check all your changed files again and update them? Thanks

Sure, I'll update the PR of the wordings you mentioned or related.

**Comment by zuiderkwast** [comment]
I'm sorry but this is starting to be very difficult to review.

The change of `REDIS_` variables is not complete. There are still variables like that, such as `REDIS_TLS_PROTO_TLSv1_2`. Don't need to do it, but write exactly what is done in the PR description.

With this kind of mass replacements, it is better to explain exactly what is done, so we can review *how it was done* instead of reading the full diff.

Also, I'm starting to think that one PR for each renamed binary would be good. At lease one commit for each binary and no revert and applied suggestion commits. Force-push is OK.

**Comment by gaod** [comment]
> I'm sorry but this is starting to be very difficult to review.
> 
> The change of `REDIS_` variables is not complete. There are still variables like that, such as `REDIS_TLS_PROTO_TLSv1_2`. Don't need to do it, but write exactly what is done in the PR description.
> 
> With this kind of mass replacements, it is better to explain exactly what is done, so we can review _how it was done_ instead of reading the full diff.
> 
> Also, I'm starting to think that one PR for each renamed binary would be good. At lease one commit for each binary and no revert and applied suggestion commits. Force-push is OK.

Yes, I think splitting the wording changes and

**Comment by gaod** [comment]
I believe the original purpose was to rename the binary, but the discussion diverged somewhat. I agree that it would be best to separate the two issues. Could the core team decide whether to handle them together in the same pull request or split them into multiple pull requests?"

**Comment by gaod** [comment]
I believe the original purpose was to rename the binary, but the discussion somewhat diverged from that. I agree that it may be better to split this into different pull requests. Could the core team decide whether to handle them together in the same pull request or to split them into multiple pull requests?

**Comment by hwware** [comment]
> I believe the original purpose was to rename the binary, but the discussion somewhat diverged from that. I agree that it may be better to split this into different pull requests. Could the core team decide whether to handle them together in the same pull request or to split them into multiple pull requests?

One big pr is not problem. Atfer review it, please run CI on your own repo and post result.  Thanks

**Comment by hwware** [comment]
> I believe the original purpose was to rename the binary, but the discussion somewhat diverged from that. I agree that it may be better to split this into different pull requests. Could the core team decide whether to handle them together in the same pull request or to split them into multiple pull requests?

Could you please first rebase unstable branch and add DCO in your commit?  Thanks

**Comment by zuiderkwast** [comment]
OK, I can accept one big PR. I prefer to skip those "Update src/module.c" and Revert commits though, so it will be easier to see only one commit for each renamed binary. You need to fix those commits without a signed-off-by anyway.

**Comment by gaod** [comment]
> > I believe the original purpose was to rename the binary, but the discussion somewhat diverged from that. I agree that it may be better to split this into different pull requests. Could the core team decide whether to handle them together in the same pull request or to split them into multiple pull requests?
> 
> One big pr is not problem. Atfer review it, please run CI on your own repo and post result. Thanks

The latest CI report is in https://github.com/gaod/valkey/actions/runs/8527284341

File: TLS.md [hunk 16]
```diff
@@ -37,21 +37,21 @@ To manually run a Redis server with TLS mode (assuming `gen-test-certs.sh` was
...
 For TLS module mode:
-    ./src/redis-server --tls-port 6379 --port 0 \
+    ./src/valkey-server --tls-port 6379 --port 0 \
         --tls-cert-file ./tests/tls/redis.crt \
         --tls-key-file ./tests/tls/redis.key \
         --tls-ca-cert-file ./tests/tls/ca.crt \
         --loadmodule src/redis-tls.so
 
-To connect to this Redis server with `redis-cli`:
+To connect to this Redis server with `valkey-cli`:
```
Comment (Shivshankar-Reddy) [inline TLS.md:52]: is this "_**To connect to this `valkey server` with `valkey-cli`**_"?

File: TLS.md [hunk 17]
```diff
@@ -37,21 +37,21 @@ To manually run a Redis server with TLS mode (assuming `gen-test-certs.sh` was
 invoked so sample certificates/keys are available):
 
 For TLS built-in mode:
-    ./src/redis-server --tls-port 6379 --port 0 \
+    ./src/valkey-server --tls-port 6379 --port 0 \
         --tls-cert-file ./tests/tls/redis.crt \
         --tls-key-file ./tests/tls/redis.key \
```
Comment (Shivshankar-Reddy) [inline TLS.md:42]: We might need to change redis.crt and redis.key to valkey files in tls mode.

Comment (zuiderkwast) [reply TLS.md:42]: Yes, but there are other PRs changing this. Let's try to avoid doing everything in one PR.

**Comment by hwware** [comment]
@gaod Hi Gaod, Thanks for your hard work first, But this pr involved too much changes and files. Could you please split it into mulltiply small pr so that we could be easier to review and merge them?  Thanks.

Note: Make sure add DCO in your commit,

File: TLS.md (hunk 17 above)
Comment (Shivshankar-Reddy) [reply TLS.md:42]: Got you!

**Comment by hwware** [comment]
Most rebranding work was done, we would like to close this PR soon.

**Pull request closed without merging**